core/ustring.cpp

```diff
@@ -3049,6 +3049,23 @@ String String::replacen(const String &p_key, const String &p_with) const {
 	return new_string;
 }
 
+String String::repeat(int p_count) const {
+
+	ERR_FAIL_COND_V_MSG(p_count < 0, "", "Parameter count should be a positive number.");
+
+	String new_string;
+	const CharType *src = this->c_str();
+	int len = length();
+
+	new_string.resize(len * p_count + 1);
```

Comment:
This is what I came up with from looking around the code (instead of allocating myself). Please review.

Reply:
The `+ 1` might not be necessary.

Reply:
I assumed so, but w/o that the last character is getting truncated in my tests. I thought it was by design (\0?).

Reply:
You probably want to get `size()` instead of `length()`

Reply:
I am getting some weird issues with size() (see below), thus using **length()+1**

![image](https://user-images.githubusercontent.com/4707543/64193939-f6ccd980-ce4b-11e9-9eae-0d7dfa93c190.png)